### ISSUE TITLE
Question: In Components v 2.0.0 package release, what versions are available at individual component level? 

### ISSUE THREAD

**Issue by lbaoyu** [Adobe-Marketing-Cloud/aem-core-wcm-components#87]
I can imagine at least one of the components would have both v1 and v2 versions available (otherwise the major version should not incremented to 2.x.x), but what would happen to the individual components that are kept intact between v1.x.x and v2.x.x package release?

**Comment by vladbailescu** [comment]
Release 2.0.0 will include:
- All existing components (version v1)
- Updated existing components with breaking changes (version v2)
- Some new components (version v1)

The objective is to be able to do in-place updates to new releases of the core components, without needing to update implementations based on older versions.

**Comment by lbaoyu** [comment]
Thanks @vladbailescu 

So the package release's major version is incremented from **N**.x.x to **N+1**.x.x whenever there is (at least one) v(N+1) version for a certain individual component become available. Of course normally there will be a bunch of them got new versions, not just one or two.

Is it a general rule that higher number version of a component can always be regarded a better version, so v(N+1) is always better than v(N)? 

Should old versions of individual components ever be declared deprecated and removed?

**Comment by vladbailescu** [comment]
TL;DR: Yes, major releases will usually bring newer components and/or newer versions for some existing components. We try to make newer versions better than the old ones.

The plan is to have:
- Micro releases (N.x.y+2) for compatible updates and improvements to the existing components
- Minor releases (N.x+1.0) for larger improvements, which are still compatible from a usage-point of view but might require a certain service pack or feature pack. These might include new versions for some of the components, although it's unlikely to have too many.
- Major releases (N+1.0.0) whenever we add more significant changes: new features, new versions for the existing components, new components.

It is very probable that most components will not get new version on each major release, especially as they become more mature.

In general we create newer component versions whenever the changes are going to be incompatible with the older markup, model or configuration options. The drivers for these changes are:
- improving how the component works
- offering more flexibility/configurability
- fixing some major issues, should it be the case
